skfuzzy/defuzzify/defuzz.py

```diff
@@ -58,7 +58,43 @@ def centroid(x, mfx):
     skfuzzy.defuzzify.defuzz, skfuzzy.defuzzify.dcentroid
 
     """
-    return (x * mfx).sum() / np.fmax(mfx.sum(),
+
+    '''
+    As we suppose linearity between each pair of points of x, we can calculate the exact area of the figure
+    (a triangle or a rectangle).
+    '''
+    sum_moment_area = 0.0
+    sum_area = 0.0
+    #If the membership function is a singleton fuzzy set:
+    if len(x)==1:
+        return x[0]*mfx[0]/np.fmax(mfx[0],
+                                     np.finfo(float).eps).astype(float)
```

Comment:
Per PEP8, any continuations within brackets/parentheticals/etc. can either be dropped directly after the bracket with an additional indent, or continued at the level of the prior bracket. E.g.,

``` python
        return x[0]*mfx[0]/np.fmax(
            mfx[0], np.finfo(float).eps).astype(float)
```

or

``` python
        return x[0]*mfx[0]/np.fmax(mfx[0], 
                                   np.finfo(float).eps).astype(float)
```

with each way equally valid.

However, as the full line does not break 80 characters (78 total), the preferred way would be

``` python
        return x[0]*mfx[0]/np.fmax(mfx[0], np.finfo(float).eps).astype(float)
```